src/connection/builders/builder_pure_rust.rs

```diff
@@ -80,11 +81,54 @@ impl PureRustConnectionBuilder {
         self
     }
 
-    /// Statement cache size. Default: 20
+    /// Connection charset. Default: UTF-8
     pub fn charset(&mut self, charset: Charset) -> &mut Self {
         self.1 = charset;
         self
     }
+
+    /// Setup the connection using the string
+    /// pattern.
+    ///
+    /// You can use the others methods(`host()`,`user()`...) to config
+    /// some default values.
+    ///
+    /// Basic string format: `firebird://{user}:{pass}@{host}:{port}/{db_name}?charset={charset}&dialect={dialect}`
+    pub fn with_string<S: Into<String>>(&mut self, s_conn: S) -> Result<&mut Self, FbError> {
```

Comment:
No need to own the string.

```suggestion
    pub fn with_string(&mut self, s_conn: &str) -> Result<&mut Self, FbError> {
```